core/src/main/java/io/aiven/klaw/service/BaseOverviewService.java

```diff
@@ -224,9 +223,19 @@ protected List<AclOverviewInfo> applyFiltersAclsForSOT(
 
     List<AclOverviewInfo> aclList = new ArrayList<>();
     AclOverviewInfo mp;
+    List<AclRequests> disableShowDeleteAcl;
 
     for (Acl aclSotItem : aclsFromSOT) {
+      disableShowDeleteAcl =
+          aclRequestsRepo.findAllByTenantIdAndEnvironmentAndRequestStatusAndTopicname(
+              tenantId,
+              aclSotItem.getEnvironment(),
+              RequestStatus.DELETED.value,
+              aclSotItem.getTopicname());
       if (aclSotItem.getAclip() != null || aclSotItem.getAclssl() != null) {
+        boolean deleteRequestExists =
+            disableShowDeleteAcl.stream()
+                .anyMatch(item -> item.getReq_no().equals(aclSotItem.getReq_no()));
```

Comment:
Why should we do extra filtration on java level?
Could we move all the predicates(filters) to sql level to minimize amount of data transferring between the app and the DB?